src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -986,10 +987,8 @@ def _create_identity_instance(cli_ctx, authority, tenant_id=None, client_id=None
 
 
 def _use_msal_managed_identity(cli_ctx):
-    # This indicates an Azure Arc-enabled server
-    from msal.managed_identity import get_managed_identity_source, AZURE_ARC
     from azure.cli.core.telemetry import set_use_msal_managed_identity
-    # PREVIEW: Use core.use_msal_managed_identity=true to enable managed identity authentication with MSAL
-    use_msal_managed_identity = cli_ctx.config.getboolean('core', 'use_msal_managed_identity', fallback=False)
+    # Use core.use_msal_managed_identity=false to use the old msrestazure implementation
+    use_msal_managed_identity = cli_ctx.config.getboolean('core', 'use_msal_managed_identity', fallback=True)
     set_use_msal_managed_identity(use_msal_managed_identity)
-    return use_msal_managed_identity or get_managed_identity_source() == AZURE_ARC
+    return use_msal_managed_identity
```

Comment:
No longer necessary to detect Azure Arc, as MSAL managed identity is now the default.

Reply:
Agree with Hang. AZURE_ARC could fail if customer accidently set `use_msal_managed_identity=false` because it's not supported with msrestazure

Reply:
Detecting Azure Arc will slow down CLI in other environments.

`use_msal_managed_identity=false` will be a hidden interface to revert to the old `msrestazure` implementation only when there are issue reported. Azure Arc users will not / need not to set it at all.